p2p/pkg/node/node.go

```diff
@@ -416,17 +417,24 @@ func NewNode(opts *Options) (*Node, error) {
 			return nil, err
 		}
 
-		validatorRouterSession := &validatorrouter.ValidatoroptinrouterCallerSession{
-			Contract: validatorRouterCaller,
-			CallOpts: bind.CallOpts{
-				From: opts.KeySigner.GetAddress(),
-			},
+		callOptsGetter := func() (*bind.CallOpts, error) {
+			blkNum, err := l1ContractRPC.BlockNumber(context.Background())
+			if err != nil {
+				return nil, err
+			}
+			currentBlkNum := big.NewInt(0).SetUint64(blkNum)
+			queryBlkNum := big.NewInt(0).Sub(currentBlkNum, big.NewInt(int64(opts.LaggeredMode)))
```

Comment:
Could we initialize opts.laggersMode as big.Int, so we could avoid casting on every api call?